### ISSUE TITLE
strange code is generated

### ISSUE THREAD

**Issue by fidokomik** [turnermm/cssperpage#2]
Bug is described in attachment because this editor bad format source code.
[bug.txt](https://github.com/turnermm/cssperpage/files/4444168/bug.txt)

What I do wrong?

**Comment by turnermm** [comment]
As stated on the plugin page:
 > The CSS must be placed in the style.css file of the plugin directory. 
There is already a style.ccss in that directory for the demo.  The css in your page has one extra closing bracket, and if you use it that way you will generate error messages.  Instead, adapt the style.css that comes with the plugin.

**Comment by fidokomik** [comment]
There is no way to write style dules directly to wiki page? I have no access to wiki directories except standard wiki admin panel. So I cannot to modify style.css file. In other words: I can install plugin "as is" only by "extensions management" and Install button. :(

**Comment by turnermm** [comment]
You can set the htmlok configuration option to true in the "Configuration Settings" on the Admin page..  This enables you to include html in your pages.   See https://www.w3schools.com/css/css_howto.asp.  What you want there is the ` internal ` style sheet.

**Comment by fidokomik** [comment]
Yes but enabling html is security hole. I mean some like this

```
~~cssp_openDIV~~
 ~~cssp_beginStyle~~
   font-size:32px;
   color:red;
~~cssp_endStyle~~
   your page data
~~cssp_closeDIV~~
```

will html generate
```
<div id="#css_per_page" style="font-size:32px; color:red;">
 your page data
</div>
```
It would be secure and simple.

**Comment by turnermm** [comment]
If you have no access to the server where you can install a style.css, htmlok is your only alternative.  There's nothing the plugin can do about it.  If this is really important to you, you may have to up your service level with your hosting provider, or find one which allows server access.

**Comment by fidokomik** [comment]
> If you have no access to the server where you can install a style.css, htmlok is your only alternative. There's nothing the plugin can do about it. If this is really important to you, you may have to up your service level with your hosting provider, or find one which allows server access.

My previous post was be an suggestion to improve your plugin. :)

My problem is different. I am wiki admin of one comunity server, but not server admin.

**Comment by turnermm** [comment]
> My previous post was be an suggestion to improve your plugin. 

 I'm afraid your suggestion is based on a misunderstanding of how dokuwiki works. Either you get access to the server or you use htmlok.

**Comment by fidokomik** [comment]
Know you this https://www.dokuwiki.org/plugin:webcode ? It works similar to my suggestion. 

And more other plugins write to server disks without having R/W access at server level. For example (bundled) Extensions manager https://www.dokuwiki.org/plugin:extension I not need to have R/W access to hosting, hosting provider have Dokuwiki preinstalled and I get access as wiki admin only. But I can install extensions.

This plugin https://www.dokuwiki.org/plugin:iframe work too very similar to my suggestion and accept parameters.

Are you willing to think about improvement if I partially rework your source and will send to you?

**Comment by turnermm** [comment]
`iframe` and `webcode` are syntax plugins that output html in place.  The extension manager is part of the dokuwiki core and has write permission because of that; otherwise it couldn't install extensions.  But since you have found the `webcode` plugin you can include your css by using that plugin and don't need `cssperpage`  to do it. 

And are you sure you don't have ftp access to your wiki?

**Comment by fidokomik** [comment]
> And are you sure you don't have ftp access to your wiki?

Yes, I am absolute sure. It is not basic hosting, it is some like corporate in house server. Person A is server manager, person B is network and security manager, person C is web administrator, person D (me) is wiki administrator, etc.Nobody of us, except person A, have direct access to hardisk.
